scripts/format-changelog.js

```diff
@@ -0,0 +1,304 @@
+        const nameB = b.split('#')[0];
+        if (nameA.toLowerCase() < nameB.toLowerCase()) return -1;
+        if (nameA.toLowerCase() > nameB.toLowerCase()) return 1;
+        return numA - numB;
+    } else {
+        // If one is a direct reference and the other is a subproject reference,
+        // the direct reference should come first
+        return isDirectA ? -1 : 1;
+    }
+}
+
+function reorderGroups(a, b) {
+    const changeTypeDiff = ChangeTypes.indexOf(a.changeType) - ChangeTypes.indexOf(b.changeType);
+    if (changeTypeDiff != 0) {
+        return changeTypeDiff;
+    }
+    // If changeType is identical, sort based on reference
+    const hasMainRefA = a.refs.length > 0 && a.refs[0].startsWith("#");
+    const hasMainRefB = b.refs.length > 0 && b.refs[0].startsWith("#");
+    // Sort by main reference number.
+    if (hasMainRefA && hasMainRefB) {
+        const numA = parseInt(a.refs[0].replace(/\D/g, ''), 10);
+        const numB = parseInt(b.refs[0].replace(/\D/g, ''), 10);
+        return numA - numB;
+    } else if (hasMainRefA && !hasMainRefB) {
+        return -1;
+    } else if (!hasMainRefA && hasMainRefB) {
+        return 1;
+    } else {
+        // Sort by reference name.
+        const nameA = a.refs.length > 0 ? a.refs[0].split('#')[0] : "";
+        const nameB = b.refs.length > 0 ? b.refs[0].split('#')[0] : "";
+        if (nameA.toLowerCase() < nameB.toLowerCase()) return -1;
+        if (nameA.toLowerCase() > nameB.toLowerCase()) return 1;
+        return 0;
+    }
+}
+
+function cleanupVersionBlock(version) {
+    // Sort the references first, numbers come before sub project refs.
+    version.entries.forEach(entry => entry.refs.sort((a, b) => referenceSort(a, b)));
+
+    // Sort the entries by change type.
+    version.entries.sort((a, b) => reorderGroups(a, b));
+
+    // Ensure entries end with a dot.
+    version.entries.forEach(entry => {
+        if (!entry.description.endsWith(".")) {
+            entry.description += ".";
+        }
+    });
+}
+
+function cleanupVersions(versions) {
+    for (const version of versions) {
+        cleanupVersionBlock(version);
+    }
+}
+
+function printChangelog(versions) {
+    for (const version of versions) {
+        console.log(`${version.version} (${version.date})`);
+        console.log(HeaderSeperator);
+        if (version.headline != "") {
+            console.log(version.headline);
+            console.log();
+        }
+        for (const entry of version.entries) {
+            var line = `- ${entry.changeType}: `;
+            if (entry.refs.length > 0) {
+                line += `[${entry.refs.join(", ")}] `;
+            }
+            line += `${entry.description}`;
+            console.log(line);
+        }
+        console.log();
+    }
+}
+
+function checkLineEndings(changelog) {
+    if (changelog.indexOf("\r\n") != -1) {
+        console.error("Changelog contains Windows line endings, please convert to Unix line endings.");
```

Comment:
Nice catch!

Reply:
Oh yeah, forgot to trigger the action testing this. Will update the first post

Reply:
Seems like git won't let me change it and I rather not bother messing around with the git settings 😄, however tested it locally just to make sure and it definitely catches that.